### ISSUE TITLE
Shadowing warning w/ latest release and MSVC

### ISSUE THREAD

**Issue by Cogitri** [chriskohlhoff/asio#535]
Hello,

after upgrading from the boost 1.73.0 to 1.74.0 MSVC issues the following warning when compiling code using `resolve_query_op`:

```
boost/asio/detail/resolve_query_op.hpp(59): warning C4459: Declaration of "query" shadows global declaration , see boost/asio/query.hpp(251): note: see declaration of "boost::asio::`anonymous-namespace'::query" 
```

**Comment by toonetown** [comment]
I am also getting this in `resolver_service.hpp`:
```
boost\asio\detail\resolver_service.hpp(75,64): warning C4459: declaration of 'query' hides global declaration
```

**Comment by chriskohlhoff** [comment]
Fixed in asio 1.18.1 / boost 1.75.
